urllib3/contrib/socks_proxy.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
```

Comment:
Well I might have to use one of those, because `socks.py` busts the import logic (I import `socks` in this module, so if I have a `socks.py` in this directory everything gets screwy.)